src/main/java/org/alfresco/service/cmr/repository/Period.java

```diff
@@ -102,6 +104,26 @@ public Period(String period)
             expression = parts[1];
         }
     }
+    
+    /**
+     * Create a period using key-value
+     * 
+     * @param source
+     */
+    public Period(Map<String, String> source)
+    {
+        if (source != null)
```

Comment:
Although code in this class such as `hashCode` appears to allow for a `null` `periodType`, it wasn't previously possible to instantiate this class without a periodType (though the `expression` is intentionally optional). This constructor allows a null `source` and will result in a `Period` with both fields `null`. Should we be throwing an `IllegalArgumentException` or similar if either the `source` map, or the `periodType` (within the source map) are null?

Reply:
yes in java doc says: A period is specified by the period type and an optional
 expression. periodType is mandatory. I will throw IllegalArgumentException. It's not possible to have a periodType null.